examples/mcu-board-support/esp32_s3_box_3.rs

```diff
@@ -0,0 +1,354 @@
+use alloc::rc::Rc;
+use core::cell::RefCell;
+use embedded_graphics_core::geometry::OriginDimensions;
+use embedded_hal::delay::DelayNs;
+use embedded_hal::digital::OutputPin;
+use embedded_hal_bus::spi::ExclusiveDevice;
+use esp_alloc as _;
+use esp_backtrace as _;
+use esp_hal::clock::CpuClock;
+use esp_hal::gpio::DriveMode;
+use esp_hal::peripherals::Peripherals;
+use esp_hal::time::Instant;
+use esp_hal::timer::timg::TimerGroup;
+use esp_hal::timer::Timer;
+use esp_hal::{
+    delay::Delay,
+    gpio::{Level, Output, OutputConfig},
+    i2c::master::I2c,
+    // init,
+    spi::master::{Config as SpiConfig, Spi},
+    spi::Mode as SpiMode,
+    time::Rate,
+};
+use esp_println::logger::init_logger_from_env;
+use gt911::Gt911Blocking;
+use log::{error, info};
+use mipidsi::options::{ColorOrder, Orientation, Rotation};
+use slint::platform::PointerEventButton;
+use slint::platform::WindowEvent;
+
+struct TimerState {
+    start_instant: Instant,
+    timer0: Option<esp_hal::timer::timg::Timer>,
+}
+
+// Provide a default implementation for TimerState.
+// We assume that Instant::EPOCH is a good default.
+impl Default for TimerState {
+    fn default() -> Self {
+        TimerState { start_instant: Instant::EPOCH, timer0: None }
+    }
+}
+
+struct EspBackend {
+    window: RefCell<Option<Rc<slint::platform::software_renderer::MinimalSoftwareWindow>>>,
+    peripherals: RefCell<Option<Peripherals>>,
+    timer_state: RefCell<TimerState>,
+}
+
+impl Default for EspBackend {
+    fn default() -> Self {
+        EspBackend {
+            window: RefCell::new(None),
+            peripherals: RefCell::new(None),
+            timer_state: RefCell::new(TimerState::default()),
+        }
+    }
+}
+
+/// Initializes the heap and sets the Slint platform.
+pub fn init() {
+    // Initialize peripherals first.
+    let peripherals = esp_hal::init(esp_hal::Config::default().with_cpu_clock(CpuClock::_240MHz));
+    init_logger_from_env();
+    info!("Peripherals initialized");
+
+    // Initialize the PSRAM allocator.
+    esp_alloc::psram_allocator!(peripherals.PSRAM, esp_hal::psram);
+
+    // Create an EspBackend that now owns the peripherals.
+    slint::platform::set_platform(Box::new(EspBackend {
+        peripherals: RefCell::new(Some(peripherals)),
+        window: RefCell::new(None),
+        timer_state: RefCell::new(TimerState::default()),
+    }))
+    .expect("backend already initialized");
+}
+
+impl slint::platform::Platform for EspBackend {
+    fn create_window_adapter(
+        &self,
+    ) -> Result<Rc<dyn slint::platform::WindowAdapter>, slint::PlatformError> {
+        let window = slint::platform::software_renderer::MinimalSoftwareWindow::new(
+            slint::platform::software_renderer::RepaintBufferType::ReusedBuffer,
+        );
+        self.window.replace(Some(window.clone()));
+        Ok(window)
+    }
+
+    fn duration_since_start(&self) -> core::time::Duration {
+        let ts = self.timer_state.borrow();
```

Comment:
You can remove all the code associated with this here, with the timing primitives from https://docs.espressif.com/projects/rust/esp-hal/1.0.0-beta.0/esp32s3/esp_hal/time/index.html

I think `core::time::Duration::from_secs(Instant::now().duration_since_epoch().as_secs())` should do it.

Reply:
Great. It works, just it was necessary to use as_millis, otherwise animations were too slow.